### PR TITLE
[bugfix] Restore module paths specified in command line when restoring module collections

### LINKED ISSUE #1631: module-path=+<path> conflict with module collections
I believe there is a conflict between `--module-path` and lmod `module restore`.
This check:

```python
import reframe as rfm
import reframe.utility.sanity as sn


@rfm.simple_test
class HelloTest(rfm.CompileOnlyRegressionTest):
    def __init__(self):
        self.valid_systems = ['pilatus:login']
        self.valid_prog_environs = ['*']
        self.sourcepath = 'hello.c'
        self.modules = ['ddt']
```

when run with:
> ~/reframe.git/bin/reframe -C ~/reframe.git/config/cscs.py -c jg.py -r -p PrgEnv-cray \
`--module-path=+/apps/eiger/UES/tools/easybuild/modules/all`

fails with (attached log):
```
  * Reason: environ error: could not execute module operation: command '/usr/share/lmod/lmod/libexec/lmod python show ddt' failed with exit code 1:
```

### PR DESCRIPTION
**Pull request by teojgo**
Fixes #1631

### CODE REVIEW

**Comment by teojgo** [comment]
@jgphpc can you check if this works for you?

File: reframe/core/modules.py [hunk 1]
```diff
@@ -729,6 +729,8 @@ def __init__(self):
                 (version, self.MIN_VERSION))
 
         self._version = version
+        self._module_paths_added = []
+        self._module_paths_removed = []
```
Comment (vkarak) [inline reframe/core/modules.py:733]: I would have a single list of tuples here of the form: `[('+', 'path/added'), ('-', '/path/removed')]`. This way you keep the sequence of additions and deletions and be completely sure that you preserve the same behaviour.

**Review by vkarak — CHANGES_REQUESTED** [review]
As for testing, you can create a temporary module collection as we do in the module collection unit tests and play with the module paths.

File: reframe/core/modules.py [hunk 2]
```diff
@@ -803,11 +806,9 @@ def emit_load_instr(self, module):
...
             # Here we append module searchpath removal/addition commands
             # since 'restore' discards previous module path manipulations
-            for mp in self._module_paths_removed:
-                instructions += [f'module unuse {mp}']
-
-            for mp in self._module_paths_added:
-                instructions += [f'module use {mp}']
+            for op, mp in self._extra_module_paths:
+                operation = 'use' if op == '+' else 'unuse'
+                instructions += [f'module {operation} {mp}']
```
Comment (vkarak) [inline reframe/core/modules.py:811]:
```suggestion
                cmds += [f'module {operation} {mp}']
```

**Review by vkarak — APPROVED** [review]
lgtm. I only have a minor cosmetic comment.

**Pull request merged**